agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/AIAgents.kt

```diff
@@ -0,0 +1,96 @@
+package ai.koog.agents.core.agent
+
+import ai.koog.agents.core.agent.GraphAIAgent.FeatureContext
+import ai.koog.agents.core.agent.config.AIAgentConfig
+import ai.koog.agents.core.agent.config.AIAgentConfigBase
+import ai.koog.agents.core.agent.entity.AIAgentGraphStrategy
+import ai.koog.agents.core.tools.ToolRegistry
+import ai.koog.prompt.dsl.prompt
+import ai.koog.prompt.executor.model.PromptExecutor
+import ai.koog.prompt.llm.LLModel
+import ai.koog.prompt.params.LLMParams
+import kotlinx.datetime.Clock
+import kotlin.reflect.typeOf
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+/**
+ * Convenience builder that creates an instance of [AIAgent], automatically deducing [GraphAIAgent.inputType] and [GraphAIAgent.outputType]
+ * from [Input] and [Output]
+ *
+ * @property promptExecutor Executor used to manage and execute prompt strings.
+ * @property strategy Strategy defining the local behavior of the agent.
+ * @property agentConfig Configuration details for the local agent that define its operational parameters.
+ * @property toolRegistry Registry of tools the agent can interact with, defaulting to an empty registry.
+ * @property installFeatures Lambda for installing additional features within the agent environment.
+ * @property clock The clock used to calculate message timestamps
+ *
+ * @see [AIAgent] class
+ */
+@OptIn(ExperimentalUuidApi::class)
+public inline fun <reified Input, reified Output> AIAgent(
```

Comment:
Now, as it's an external function it would be even harder to instantiate from Java.

Maybe let's keep everything inside `AIAgent` and make them separate constructors (and in Java they would be factory methods) ?